pkg/controllers/management/clusterprovisioner/provisioner.go

```diff
@@ -287,6 +288,34 @@ func (p *Provisioner) reconcileCluster(cluster *v3.Cluster, create bool) (*v3.Cl
 		apiEndpoint, serviceAccountToken, caCert, err = p.driverUpdate(cluster, *spec)
 	} else {
 		logrus.Infof("Updating cluster [%s]", cluster.Name)
+
+		v3.ClusterConditionUpdated.Unknown(cluster)
+		cluster, err := p.Clusters.Update(cluster)
+		if err != nil && apierrors.IsConflict(err) {
+			for i := 0; i < 4; i++ {
+				if cluster, err = p.Clusters.Get(cluster.Name, metav1.GetOptions{}); err != nil {
+					if apierrors.IsConflict(err) {
```

Comment:
we won't get Conflict error on API GET, we should just return error in this case